lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -286,11 +286,10 @@ void Visit(ParseNode *pnode, ByteCodeGenerator* byteCodeGenerator, PrefixFn pref
     }
     case knopClassDecl:
     {
-        // Visit the extends expression first, since it's bound outside the scope containing the class name.
-        Visit(pnode->sxClass.pnodeExtends, byteCodeGenerator, prefix, postfix);
         Visit(pnode->sxClass.pnodeDeclName, byteCodeGenerator, prefix, postfix);
         // Now visit the class name and methods.
         BeginVisitBlock(pnode->sxClass.pnodeBlock, byteCodeGenerator);
+        Visit(pnode->sxClass.pnodeExtends, byteCodeGenerator, prefix, postfix);
```

Comment:
We might want to add a comment about the reason for the visitation order since we removed the old comment.